### DISCUSSION TITLE
Max number of devices

### DISCUSSION THREAD

**Discussion by MadMirza** [Koenkk/zigbee2mqtt#17830] [General]
I am planning on doing a Zigbee network of around 400-500 devices. I searched around to find out if Z2M has a max number of devices over which no more devices can be added.  I could not find a solid answer.
I have personally seen a network work stably with around 200 devices.
Should I be worried about the huge number of devices or having enough number of routers would help to keep the network stable and add all the devices in? 
If this is possible what could be some best practices that I could follow to make the best out of my zigbee network?

Thanks :)

**Comment by hdejongh** [comment]
Hi @MadMirza 

Did you ever go through with you zigbee network cause I have the same question so I was hoping to hear from you how it went:)

Thanks!

**Reply by MadMirza** [reply]
Hey @hdejongh, just saw the notification!

The project was delayed quite a bit, i ended up with only 218 devices of which 100+ are routing capable devices, using HA Connect ZBT-1 (formerly SkyConnect).

Initial pairing and configs took me a few days.
I paired everything from nearer from coordinator to farthest, only routing devices were added in first, then all endDevices ( had minor hiccups pairing in, multiple retries were required for some).

It works to this day, 99%~ of the time things are good. There are minor delays at times, which we got used to. Sometimes a few devices shows offline when unused for long. 

Most of the devices are Aqara, Tuya, Ikea, Xiaomi.

Once power outage caused system to offline and upon coming back online, I had to re-pair a few devices and took a few hours until the Zigbee network was okay.
